### PR TITLE
CMR-8749: Use individual concept types for associations in variable search

### PR DESCRIPTION
**Pull request by sxu123**

### CODE REVIEW

File: indexer-app/src/cmr/indexer/data/concepts/service.clj [hunk 1]
```diff
@@ -58,13 +58,15 @@
...
-                                   concept-id))))})))
+       :associations-gzip-b64 (when (or (seq service-associations)
+                                        (seq generic-associations))
+                                (util/string->gzip-base64
+                                 (pr-str
+                                  (util/remove-map-keys
+                                   empty?
+                                   (assoc-util/assoc-list->assoc-struct
+                                    (concat service-associations generic-associations)
+                                     concept-id)))))})))
```
Comment (jaybarra) [inline indexer-app/src/cmr/indexer/data/concepts/service.clj:69]: Would recommend simplification with thrush, and concat the associations higher up into a single list.
```suggestion
       :associations-gzip-b64 (when (seq any-assocs)
                                        (->> concept-id 
                                                 (assoc-util/assoc-list->assoc-struct any-assocs)
                                                 (util/remove-map-keys empty?)
                                                 pr-str
                                                 util/string->gzip-base64)
```

File: indexer-app/src/cmr/indexer/data/concepts/variable.clj [hunk 2]
```diff
@@ -92,25 +84,15 @@
...
-                                                                 (:source-revision-id %))
-                                                               generic-associations)
-                                                          (map #(get-concept-id-revision-id-map
-                                                                 (:associated-concept-id %)
-                                                                 (:associated-revision-id %))
-                                                               generic-associations))]
-                                            (remove #(= concept-id (:concept-id %)) generics))}))))})))
+       :associations-gzip-b64 (when (or (seq variable-associations)
+                                        (seq generic-associations))
+                                (util/string->gzip-base64
```
Comment (jaybarra) [inline indexer-app/src/cmr/indexer/data/concepts/variable.clj:89]: Would recommend using `->>` for simplification here as well

**Review by jaybarra — APPROVED** [review]

**Review by jceaser — APPROVED** [review]

File: umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json [hunk 3]
```diff
@@ -67,34 +67,181 @@
...
     },
-    "AssociationType": {
+    "AssociationsInItemType": {
       "type": "object",
       "additionalProperties": false,
-      "description": "Concepts associated with the variable.",
+      "description": "Any concept associated with the service.",
       "properties": {
+        "variables": {
+          "description": "The concept ids of variables assciated with the service",
```
Comment (zimzoom) [inline umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json:88]: typo

File: umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json [hunk 4]
```diff
@@ -67,34 +67,181 @@
...
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConceptRevisionType"
+          },
+          "minItems": 1
+        },
         "collections": {
-          "description": "The collections associated with the variable.",
+          "description": "The concept ids of collections assciated with the service",
```
Comment (zimzoom) [inline umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json:96]: typo

File: umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json [hunk 5]
```diff
@@ -67,34 +67,181 @@
...
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConceptRevisionType"
+          },
+          "minItems": 1
+        },
+        "tools": {
+          "description": "The concept ids of tools assciated with the service",
```
Comment (zimzoom) [inline umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json:104]: typo

File: umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json [hunk 6]
```diff
@@ -67,34 +67,181 @@
...
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConceptRevisionType"
+          },
+          "minItems": 1
+        },
+        "dataqualitysummaries": {
+          "description": "The concept ids of data quality summary concepts assciated with the service",
```
Comment (zimzoom) [inline umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json:112]: typo

File: umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json [hunk 7]
```diff
@@ -67,34 +67,181 @@
...
           "type": "array",
           "items": {
             "$ref": "#/definitions/ConceptRevisionType"
           },
           "minItems": 1
         },
-        "generics": {
-          "description": "The generic concepts associated with the variable.",
+        "orderoptions": {
+          "description": "The concept ids of order option concepts assciated with the service",
```
Comment (zimzoom) [inline umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json:120]: typo

File: umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json [hunk 8]
```diff
@@ -67,34 +67,181 @@
...
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConceptRevisionType"
+          },
+          "minItems": 1
+        },
+        "serviceoptions": {
+          "description": "The concept ids of service option concepts assciated with the service",
```
Comment (zimzoom) [inline umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json:128]: typo

File: umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json [hunk 9]
```diff
@@ -67,34 +67,181 @@
...
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConceptRevisionType"
+          },
+          "minItems": 1
+        },
+        "serviceentries": {
+          "description": "The concept ids of service entry concepts assciated with the service",
```
Comment (zimzoom) [inline umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json:136]: typo

File: umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json [hunk 10]
```diff
@@ -67,34 +67,181 @@
...
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConceptRevisionType"
+          },
+          "minItems": 1
+        },
+        "grids": {
+          "description": "The concept ids of grid concepts assciated with the service",
```
Comment (zimzoom) [inline umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json:144]: typo

File: umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json [hunk 11]
```diff
@@ -67,34 +67,181 @@
...
-      "required": ["collections"]
+      }
+    },
+    "AssociationsType": {
+      "type": "object",
+      "additionalProperties": false,
+      "description": "Any concept associated with the service.",
+      "properties": {
+        "variables": {
+          "description": "The concept ids of variables assciated with the service",
```
Comment (zimzoom) [inline umm-spec-lib/resources/json-schemas/variable/umm/v1.8.1/umm-var-search-results-json-schema.json:159]: typo, you can probably just do find and replace on string "assciated" for all of these.

**Review by zimzoom — APPROVED** [review]

**Pull request merged**